Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,335 @@ public static class BalanceStats
+            return richestPeople.ToString();
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal min = decimal.MaxValue;
+            string result = "";
+            StringBuilder poorPeople = new StringBuilder();
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                string value = auxBalances[^1..][0];
+                decimal.TryParse(value, out decimal currentBalance);
+                if (min > currentBalance)
+                {
+                    min = currentBalance;
+                    poorPeople.Clear();
+                    poorPeople.Append(auxBalances[0]);
+                    result = $" has the least money. {min:C0}.";
+                    moreThanOne = false;
+                }
+                else if(min.Equals(currentBalance))
+                {
+                    poorPeople.Append($", {auxBalances[0]}");
+                    result = $" have the least money. {min:C0}.";
+                    moreThanOne = true;
+                }
+            }
+
+            if (moreThanOne)
+            {
+                ReplaceLastCommaWithAnd(poorPeople);
+            }
+            poorPeople.Append(result);
+            return poorPeople.ToString();
+        }
+        /// <summary>
+        /// Makes a customizable invariant culture without currencygroupseparator and currencynegativepattern = 1
+        /// </summary>
+        /// <returns>an Invariant Culture CultureInfo</returns>
+        private static CultureInfo GetCustomInvariantCulture()
+        {
+            CultureInfo cult = (CultureInfo)CultureInfo.InvariantCulture.Clone();
+            NumberFormatInfo nfi = cult.NumberFormat;
+            nfi.CurrencyNegativePattern = 1;
+            nfi.CurrencyGroupSeparator = "";
+
+            return cult;
+        }
+        /// <summary>
+        /// verify if string array is null or empty
+        /// </summary>
+        /// <param name="array"></param>
+        /// <returns></returns>
+        private static bool IsStrArrayNullOrEmpty(string[] array) 
+        {
+            return (array == null || array.Length == 0) ? true : false;
+        }
+        /// <summary>
+        /// verify if decimal array is null or empty
+        /// </summary>
+        /// <param name="array"></param>
+        /// <returns></returns>
+        private static bool IsDecArrayNullOrEmpty(decimal[] array)
+        {
+            return (array == null || array.Length == 0) ? true : false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)